### PR TITLE
Create 2020-07-12-article-blazor-compelling-example.md

### PR DESCRIPTION
**Pull request by richbryant**
added blog article about RxUI and Blazor

**What are the main goals of this change?**
- [ ] Better readability (style, grammar, spelling...)
- [ ] Content correction (accuracy, wrong information...)
- [X] New content

**Is this change related to any reported issue? (Optional)**
nope

**Notes (Optional)**
None.  Plugs language-ext because that's how I write code.

### CODE REVIEW

File: input/blog/2020-07-12-article-blazor-compelling-example.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+As you can see, it has text input, it does pipelines, it has data, it has views and viewmodels.... Why not, I thought? What are the steps?  
+  
+1. Follow the instructions Rich, you idiot
+
+Yeah, that makes sense.  Step 1 - build the Compelling Example just like the instructions say except that I used WPF in .NET Core 3.1 which worked flawlessly and I didn't have to change anything.  I mostly did this because Blazor is a .NET Core app and I wanted to avoid any hassles later.  The screenshot above is my working WPF project.
+
+2. Add a Blazor project.  
+
+I used Blazor AspNetCore Hosted for reasons which I hope will become clear later and I moved the actual "get the data" stuff to a service inside the Server project.  I also created a ViewModels project (a .NET Standard 2.1 clas library) because I don't want to copy my ViewModels, I want them shared.
```
Comment (RLittlesII) [inline input/blog/2020-07-12-article-blazor-compelling-example.md:31]: a .NET Standard 2.1 class library

Comment (richbryant) [reply input/blog/2020-07-12-article-blazor-compelling-example.md:31]: Will fix that in the morning. Its 0120hrs here.

File: input/blog/2020-07-12-article-blazor-compelling-example.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+  
+I hope by now people are starting to learn about the existence of the [ReactiveUI.Blazor](https://www.nuget.org/packages/ReactiveUI.Blazor/) package.  Blazor means "writing Single Page Apps like Angular or Vue in C#" and it has all the potential in the world.  And ReactiveUI supports it.  The official [samples are here](https://github.com/reactiveui/ReactiveUI.Samples/tree/main/blazor) but I don't feel that they really sell the reason _why_ you'd want to do web work with ReactiveUI, and I should know, I wrote them.
+  
+For me, the big seller is that you've already got a WPF or Xamarin or WinForms app using ReactiveUI and you want a web version.  In such a case, you can rewrite everything from scratch _or_ you can use Blazor and keep the logic in your ViewModels absolutely unchanged.  Sounds nice in theory but the web is a tricky thing, so does it actually work?  I decided to try it.  
+  
+### The ReactiveUI Compelling Example  
+  
+The ReactiiveUI website Getting Started section links a [Compelling Example](https://www.reactiveui.net/docs/getting-started/compelling-example), a simple Nuget browser that you can build with copypasta in about 5 minutes.
```
[the remark this answers is not in the document]
Comment (richbryant) [reply input/blog/2020-07-12-article-blazor-compelling-example.md:19]: Clarify please?

**Comment by glennawatson** [comment]
If you don't mind let's focus on this going into a file under getting started

**Comment by richbryant** [comment]
> If you don't mind let's focus on this going into a file under getting started

Getting Started needs something else to focus on platforms.  This is about porting, not about getting started.  It could have been any RxUI app, I just picked on that already has a build walkthrough.

**Comment by glennawatson** [comment]
Let's readdress this once we have a compelling example for the user to work against. Seems this article is fighting a bit with that. Eg if you can reference a getting started example it'd make it flow a lot better

**Comment by richbryant** [comment]
> Let's readdress this once we have a compelling example for the user to work against. Seems this article is fighting a bit with that. Eg if you can reference a getting started example it'd make it flow a lot better

I don't quite understand, to be honest.  This is a blog about porting a WPF/RxUI app to the web.  As I said, it could have been any app.  I agree that the Getting Started section of the website would benefit from notes about Blazor, Xamarin and others but I don't think that's urgent.  After all, it's been _just WPF_ for years, even while WinForms works and Xamarin works and...

Well, you get the idea.  The stuff in here is not official documentation material and was never intended to be.  It's just a blog about porting some code.

**Comment by glennawatson** [comment]
Lets get your article in and then we'll worry about getting the `getting started` section to include a blazor (and other platform) examples.

**Pull request merged**